packages/transpiler/src/transformations/fix-import-directives.ts

```diff
@@ -0,0 +1,20 @@
+import { getImportDirectives, getSourceIndices } from '../solc/ast-utils';
+import { Artifact } from '../solc/artifact';
+import { Transformation } from '../transformation';
+
+export function fixImportDirectives(artifact: Artifact, artifacts: Artifact[], contracts: string[]): Transformation[] {
+  const imports = getImportDirectives(artifact.ast);
+  return imports.map(imp => {
+    const [start, len] = getSourceIndices(imp);
+    const isTranspiled = artifacts.some(
+      art => art.ast.id === imp.sourceUnit && contracts.some(contract => contract === art.contractName),
+    );
+    const prefix = !imp.file.startsWith('.') ? './' : '';
+    const fixedPath = `import "${prefix}${imp.file.replace('.sol', 'Upgradable.sol')}";`;
+    return {
+      start,
+      end: start + len,
+      text: !isTranspiled ? `import "${imp.absolutePath}";` : fixedPath,
```

Comment:
This will not work for complex imports like `import { A } from './A.sol'`. Ideally we would return a transformation that only changes the path string of the import. I would've expected the AST to contain the path string as an AST node but I don't see it.

Additionally, in the case that the import is not transpiled, is it possible to not apply a transformation at all?

Reply:
Yep. Completely missed `import` syntax. It is also possible to do.
 ```Solidity
import * as symbolName from "filename";
import {symbol1 as alias, symbol2} from "filename";
import "filename" as symbolName;
```

>  I would've expected the AST to contain the path string as an AST node but I don't see it.
It is a property `file` in `ImportDirective`

```JSON
{
        "absolutePath": "contracts/DiamondInheritance.sol",
        "file": "./DiamondInheritance.sol",
        "id": 985,
        "nodeType": "ImportDirective",
        "scope": 1019,
        "sourceUnit": 879,
        "src": "290:60:10"
)
```

Reply:
There are also other edge cases, for example:
```Solidity
pragma solidity ^0.5.0;

import * as ERC20 from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
import {ERC20Detailed as Detailed} from "@openzeppelin/contracts/token/ERC20/ERC20Detailed.sol";
import "@openzeppelin/contracts/math/Math.sol" as math;
import { DA, DB1, DB2, DC } from "./DiamondInheritance.sol";
import "@openzeppelin/contracts/math/SafeMath.sol";

contract Imports is ERC20.ERC20, Detailed, DC {
    constructor(uint256 initialSupply) public Detailed("Gold", "GLD", 18) {
        uint finalSupply = math.Math.max(0, initialSupply);
        _mint(msg.sender, SafeMath.add(finalSupply, 10));
    }
}
``` 

The whole set of import issues seems big enough to be a separate PR.